workspace/workspace_test.go

```diff
@@ -5,11 +5,88 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"sort"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+func TestWorkspacePotentialExercises(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "walk")
+	defer os.RemoveAll(tmpDir)
+	assert.NoError(t, err)
+
+	a1 := filepath.Join(tmpDir, "track-a", "exercise-one")
+	b1 := filepath.Join(tmpDir, "track-b", "exercise-one")
+	b2 := filepath.Join(tmpDir, "track-b", "exercise-two")
+
+	// It should ignore other people's exercises.
+	alice := filepath.Join(tmpDir, "users", "alice", "track-a", "exercise-one")
+
+	// It should ignore nested dirs within exercises.
+	nested := filepath.Join(a1, "subdir", "deeper-dir", "another-deep-dir")
+
+	for _, path := range []string{a1, b1, b2, alice, nested} {
+		err := os.MkdirAll(path, os.FileMode(0755))
+		assert.NoError(t, err)
+	}
+
+	ws, err := New(tmpDir)
+	assert.NoError(t, err)
+
+	exercises, err := ws.PotentialExercises()
+	assert.NoError(t, err)
+	if assert.Equal(t, 3, len(exercises)) {
+		paths := make([]string, len(exercises))
+		for i, e := range exercises {
+			paths[i] = e.Path()
+		}
+
+		sort.Strings(paths)
+		assert.Equal(t, paths[0], "track-a/exercise-one")
+		assert.Equal(t, paths[1], "track-b/exercise-one")
+		assert.Equal(t, paths[2], "track-b/exercise-two")
+	}
+}
+
+func TestWorkspaceExercises(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "walk-with-metadata")
+	defer os.RemoveAll(tmpDir)
+	assert.NoError(t, err)
+
+	a1 := filepath.Join(tmpDir, "track-a", "exercise-one")
+	a2 := filepath.Join(tmpDir, "track-a", "exercise-two") // no metadata
+	b1 := filepath.Join(tmpDir, "track-b", "exercise-one")
+	b2 := filepath.Join(tmpDir, "track-b", "exercise-two")
+
+	for _, path := range []string{a1, a2, b1, b2} {
```

Comment:
Out of curiousity is there a reason why you went with two loops with two slices as opposed to having one slice, one loop with a guard clause for `path == a2`

Reply:
No, I didn't have a good reason. It was the first thing I thought of (and I was copy/pasting pretty heavily). I think I like the guard approach better. I'll update that.